algorithms/src/snark/marlin/ahp/ahp.rs

```diff
@@ -189,26 +189,73 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
         let delta = state.second_round_message.unwrap().delta;
         let beta = state.third_round_message.unwrap().beta;
         let gamma = state.gamma.unwrap();
+        let zeta = state.zeta.unwrap();
 
         let mut linear_combinations = BTreeMap::new();
 
         // Lincheck sumcheck:
+        let z_a_s = (0..state.batch_size)
+            .map(|i| {
+                let z_a_i = witness_label("z_a", i);
+                LinearCombination::new(z_a_i.clone(), [(F::one(), z_a_i)])
+            })
+            .collect::<Vec<_>>();
```

Comment:
do these need to be collected? seems like they get iterated over immediately afterwards

Reply:
ah, I see that they are later reused too; there's still a possibility the allocations can be avoided, but it's not as trivial, so might be best to leave it for later :+1: